### PR TITLE
user: uses invenio access profile

### LINKED ISSUE #3405: Change password view: problem with certain characters
### Bug description:
Trying to change a password with an invalid character raises an error in flask_security `UnicodeDecodeError` and seems to reset the password or leave it empty, preventing to log in again. (See Sentry RERO-ILS-33R).

### Expected behavior:
If I try to change my password to an invalid character that will prevent unicode ASCII decoding, the interface should warn me and prevent me to click save, or at least cancel the password change function and keep the previous password.

### Steps to Reproduce:
1. Log in with any user
2. Click "change my password"
3. Enter a new password containing for example character "ë"
4. See error and see that there is no way to login again with either previous or new password

### Context

`v1.17.0`

### LINKED ISSUE #3441: Upgrade to Flask>2.0.0
### How it works
More and more dependencies need Flask>2.0.0 (invenio-sip2, flask-wiki, etc) to work with rero-ils. We should upgrade the app to this version.

### Improvement suggestion
- `invenio-user-profile`: stop using our own fork and go back to the official invenio release that is compatible with flask2. Adapt RERO ILS to work with that version. Our *hacks* should now be possible natively in the newest version. 
- `invenio-circulation`: We are waiting for a release of this package with flask 2.0.0 compatibility. It has been singaled to invenio.
- Update all dependencies
    - `invenio-sip2`
    - `flask-wiki`
    - `rero-invenio-base` (first publish a version)

### PR DESCRIPTION
**Pull request by jma**
* Replaces invenio-userprofile by the invenio access profile.
* Updates several packages.
* Moves `login_user_via_session` fonction in the tests as some problems
  occurs when a session merge is called between this function and the http
  request.
* Removes useless babel mock in the tests.
* Moves username from the invenio userprofile to the main user database.
* Adds a record constructor for a record `replace_refs` as it is not
  done anymore by invenio.
* Adds tests to check the existance of `_updated` and `_created` in the
  document index.
* Removes document types for elasticsearch operations.
* Adds marshmalow schema for user profile.
* Closes #3441.
* Closes #3405.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

### CODE REVIEW

**Comment by jma** [comment]
Sorry for the large PR. I keep this into two commits.

**Review by zannkukai — APPROVED** [review]
I didn't review any `tests` files. Huge and great job !

File: rero_ils/modules/api.py [hunk 1]
```diff
@@ -553,6 +550,13 @@ def get_metadata_identifier_names(cls):
         identifier = cls.provider.identifier
         return metadata, identifier
 
+    def replace_refs(self):
+        """Replace the ``$ref`` keys within the JSON.
+
+        Note: needed for jsonref>=1.0.0
+        """
+        return (type(self))(super().replace_refs())
```
[the remark this answers is not in the document]
Comment (jma) [reply rero_ils/modules/api.py:558]: Yes you are right. It was the case in old version of invenio. We have to do this for legacy compatibility. We do not have any recursion except if the constructor call replace_refs() which is not the case.

**Review by rerowep — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -57,63 +84,45 @@ xmltodict = "*"
...
 DeepDiff = "^5.5.0"
 docutils = "<0.18.0"
 wtforms = "<3.0.0"
-flask-wtf = "<1.0.0"
-# TODO: ImportError: cannot import name 'soft_unicode' from 'markupsafe'
-markupsafe = "<2.1.0"
 poethepoet = "^0.12.3"
-flask-wiki = "^0.2.2"
-pytest-invenio = ">=1.4.0,<1.4.12"
+flask-wiki = { git = "https://github.com/rero/flask-wiki.git", tag = "v0.3.0" }
```
Comment (PascalRepond) [inline pyproject.toml:109]: `flask-wiki = "^0.3.0"`
v0.3.0 is published on pypi

**Review by PascalRepond — APPROVED** [review]

**Review by lauren-d — APPROVED** [review]

**Pull request merged**